tests/aggregation/tests.py

```diff
@@ -1190,6 +1190,21 @@ def test_aggregation_subquery_annotation_values(self):
             },
         ])
 
+    @skipUnlessDBFeature('supports_subqueries_in_group_by')
+    def test_aggregation_subquery_annotation_multivalued(self):
+        """
+        Subquery annotations must be included in the GROUP BY if they use
+        potentially multivalued relations (contain the LOOKUP_SEP).
+        """
+        subquery_qs = Author.objects.filter(
+            pk=OuterRef('pk'),
+            book__name=OuterRef('book__name'),
+        ).values('pk')
+        author_qs = Author.objects.annotate(
+            subquery_id=Subquery(subquery_qs),
+        ).annotate(count=Count('book'))
+        self.assertEqual(author_qs.count(), Author.objects.count())
```

Comment:
This tests fails on MySQL 5.7+ even before fb3f034f1c63160c0ff13c609acd01c18be12f80. It is related with collapsing `GROUP BY` only to the main table's primary key on MySQL.

Reply:
@felixxm that's a tricky one for sure. We could adjust MySQL's `allows_group_by_pk` feature to be based of `not ONLY_FULL_GROUP_BY` but that would likely incur a large performance hit which is definitely not suitable for a backport. I guess we could always skip the test on MySQL for now.

Reply:
Yes, I will add a separate ticket for this, leave a ref here, and skip this test on MySQL (it works on MariaDB). Thanks.

Reply:
Done, see ticket-31331. I decided to skip this test also on MariaDB, because `ONLY_FULL_GROUP_BY` mode can be enabled on this database.